CHANGELOG.md

```diff
@@ -5,6 +5,7 @@
 #### Bugs
 * Fix #4369: Informers will retry with a backoff on list/watch failure as they did in 5.12 and prior.
 * Fix #4350: SchemaSwap annotation is now repeatable and is applied multiple times if classes are used more than once in the class hierarchy.
+* Fix #4426: [java-generator] Encode a `JsonNode` instead of an Object if `x-kubernetes-preserve-unknown-fields` is present and the type is null.
```

Comment:
I aligned the handling of this situation in light of the better understanding of the spec:
https://github.com/fabric8io/kubernetes-client/pull/4429/commits/b9f5d4d19305f5967fca80aac9d015b3e50af188

The missing part was to remove the `type` in the case of `JsonNode` (it was `object`).

Reply:
Thanks for addressing my comment, @andreaTP. I was actualy more concerned about tracking it by oggin a new issue, than addressing it here. 
For sure, the CRD generator would be aligned if we let the change go in, but I see the checks about test coverage and duplication aren't passing after it was added. 
Hence, the decision here seems to be about whether to keep this change, and make the CI checks pass, or to add it to a separate PR with its own tracker issue. 

This could also be the opportunity for discussing and defining a best practice for the upcoming CRD-gen and Java-gen fixes and enhancements.

WDYT?

Reply:
The SonarCloud checks should be taken "cum grano salis" and is not a mandatory check for merging 🙂

Reply:
Right then! :rocket: